### PR TITLE
WSTG-IDNT-01 Role Definitions Rewrite

### PR DESCRIPTION
**Pull request by ThunderSon**
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Rewrite the scenario to provide better guidance
- Better sectioning to the scenario

Points of discussion:
- Links to other test cases
- Access Control is heavily impacted by this. Did I cross any boundaries with it, or is this purely roles definitions?
- I removed the remediation section as it's straightforward and as everything is well detailed.

### CODE REVIEW

File: document/4-Web_Application_Security_Testing/03-Identity_Management_Testing/01-Test_Role_Definitions.md [hunk 1]
```diff
@@ -6,52 +6,60 @@
...
+After gaining access to the roles on the system, the tester must understand the permissions provided to each role.
 
-1. Login as administrator
-2. Visit admin page. i.e. `http://targetSite/Admin`
-3. Logout
-4. Login as normal user
-5. Visit the admin page URL directly. `http://targetSite/Admin`
+A support engineer shouldn't be able to conduct administrator functionalities, manage the backups, or conduct any transactions in the place of a user.
+
+An administrator shouldn't have full powers on the system and should have a maker-checker principle, or use MFA to ensure that the administrator is conducting the transaction. A clear example on this was the [Twitter incident in 2020](https://blog.twitter.com/en_us/topics/company/2020/an-update-on-our-security-incident.html).
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/03-Identity_Management_Testing/01-Test_Role_Definitions.md:51]:
```suggestion
An administrator shouldn't have full powers on the system. Sensitive admin functionality should leverage a maker-checker principle, or use MFA to ensure that the administrator is conducting the transaction. A clear example on this was the [Twitter incident in 2020](https://blog.twitter.com/en_us/topics/company/2020/an-update-on-our-security-incident.html).
```

File: document/4-Web_Application_Security_Testing/03-Identity_Management_Testing/01-Test_Role_Definitions.md [hunk 2]
```diff
@@ -6,52 +6,60 @@
...
 
 ## Tools
 
-While the most thorough and accurate approach to completing this test is to conduct it manually, [spidering tools](https://www.zaproxy.org/docs/desktop/start/features/spider/) are also useful. Log on with each role in turn and spider the application (don't forget to exclude the logout link from the spidering).
+The above mentioned tests can be conducted without the use of any tool, except the one being used to access the system.
+
+To make things easier and more documented, one can use:
+
+- [Burp's Autorize extension](https://github.com/Quitten/Autorize)
+- [ZAP's spider](https://www.zaproxy.org/docs/desktop/start/features/spider/)
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/03-Identity_Management_Testing/01-Test_Role_Definitions.md:60]:
```suggestion
- [ZAP's Access Control Testing add-on](https://www.zaproxy.org/docs/desktop/addons/access-control-testing/)
```

**Review by kingthorin — CHANGES_REQUESTED** [review]
Just minor tweaks. Nice work!

File: document/4-Web_Application_Security_Testing/03-Identity_Management_Testing/01-Test_Role_Definitions.md [hunk 3]
```diff
@@ -6,52 +6,60 @@
...
-| Manager | Read | Customer records |  Only records related to business unit |
-|  Staff | Read | Customer records |Only records associated with customers assigned by Manager |
-|Customer | Read | Customer record | Only own record |
+After identifying possible attack vectors, the tester needs to test and validate that they can access the available roles.
 
-A real world example of role definitions can be found in the [WordPress roles documentation](https://wordpress.org/support/article/roles-and-capabilities/). WordPress has six default roles ranging from Super Admin to a Subscriber.
+> Some applications define the roles of the user on creation, through rigirous checks and policies, or by ensuring that the user's role is properly protected through a signature created by the backend. Finding that roles exist doesn't mean that they're a vulnerability.
 
-### Example 2
+### Review Roles Permissions
```
[the remark this answers is not in the document]
Comment (ThunderSon) [reply document/4-Web_Application_Security_Testing/03-Identity_Management_Testing/01-Test_Role_Definitions.md:45]: Plural. You have multiple roles with multiple permissions.

**Comment by ThunderSon** [comment]
There's only the header. Let me know what works best for it.

File: document/4-Web_Application_Security_Testing/03-Identity_Management_Testing/01-Test_Role_Definitions.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kingthorin) [reply document/4-Web_Application_Security_Testing/03-Identity_Management_Testing/01-Test_Role_Definitions.md:45]: It reads awkwardly, I'll let it go for now.

**Review by kingthorin — APPROVED** [review]

**Pull request merged**